src/_png.cpp

```diff
@@ -274,15 +269,21 @@ static PyObject *Py_write_png(PyObject *self, PyObject *args, PyObject *kwds)
             PyErr_SetString(PyExc_TypeError, "metadata must be a dict or None");
             goto exit;
         }
-        meta_size = PyDict_Size(metadata);
-        text = new png_text[meta_size];
+
+        Py_ssize_t meta_size = PyDict_Size(metadata);
+        png_text *text = new png_text[meta_size];
+        PyObject *meta_key, *meta_val;
+        Py_ssize_t pos = 0;
+        int meta_pos = 0;
+        std::vector<PyObject*> temp_strs;
```

Comment:
don't know if it is worth it, but when doing push_back()s to a vector, one can save a fair amount of memory and allocation time by doing `temp_strs.reserve(meta_size);` before the loop for strings since we know the upper limit for this vector.

https://en.cppreference.com/w/cpp/container/vector/reserve

Reply:
I expect it won't be more than 10 or maybe 20, but I've added it since we do know the limit.